lib/fr-scanedit.cpp

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright (C) 2015 Yubico AB.  All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are
+met:
+
+   1. Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+
+   2. Redistributions in binary form must reproduce the above
+      copyright notice, this list of conditions and the following
+      disclaimer in the documentation and/or other materials provided
+      with the distribution.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "fr-scanedit.h"
+
+#define	SHIFT   0x80
+
+FrScanEdit::FrScanEdit() {
+    /* Generated using: loadkeys -m fr-pc | python convert.py */
+    /* https://gist.github.com/mickael9/567ce26b028ce89cafcaf4250464c526#file-convert-py */
+    keyMap = QMap<int, int>();
+    keyMap[0x09] = 0x2b; /* '\t' */
+    keyMap[0x0a] = 0x28; /* '\n' */
+    keyMap[0x1b] = 0x29; /* '\x1b' */
```

Comment:
Since "esc" (0x1b) is not a printable char, it shouldn't be present in a password.